packages/types/src/identity/Identity.ts

```diff
@@ -0,0 +1,10 @@
+export interface Identity {
+  /**
+   * A {Date} when the identity or credential will no longer be accepted.
+   */
+  readonly expiration?: Date;
+}
+
+export type IdentityProvider<IdentityT extends Identity> = {
+  (identityProperties?: Record<string, any>): Promise<IdentityT>;
+};
```

Comment:
Does it matter if `IdentityProvider` is a type or an interface?

For reference, `Provider` is an interface: https://github.com/aws/aws-sdk-js-v3/blob/main/packages/types/src/util.ts#L34-L43.